datastacks/utils.py

```diff
@@ -0,0 +1,74 @@
+import click
+import yaml
+
+from pydantic import BaseModel
+from datastacks.config_class import IngestConfig
+from jinja2 import Environment, FileSystemLoader
+from pathlib import Path
+
+
+def render_template_components(
+    config: BaseModel, template_source_path: str, target_dir: str
+) -> None:
+    """Renders all templates within a given path with provided config, and saves results into a new target path,
+    while maintaining folder structure and removing jinja file extensions.
```

Comment:
Consider adding the information that any pre-existing files are overwritten. Feel free to resolve.

Reply:
Done